client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/ClinicalDetailsSchema.ts

```diff
@@ -82,32 +82,32 @@ const ClinicalDetailsSchema = yup.object().shape({
     [ClinicalDetailsFields.IS_ISOLATION_PROBLEM]: yup.boolean().required(),
     [ClinicalDetailsFields.IS_ISOLATION_PROBLEM_MORE_INFO]: yup.string().when(
         ClinicalDetailsFields.IS_ISOLATION_PROBLEM, {
-        is: true,
-        then: yup.string().required(requiredText),
-        otherwise: yup.string()
-    }),
+            is: true,
+            then: yup.string().required(requiredText),
+            otherwise: yup.string()
+        }),
     [ClinicalDetailsFields.DOES_HAVE_SYMPTOMS]: yup.boolean().required(),
-    [ClinicalDetailsFields.SYMPTOMS_START_DATE]: yup.date().when(
-        ClinicalDetailsFields.DOES_HAVE_SYMPTOMS, {
-        is: true,
-        then: yup.date().required(requiredText).typeError(requiredText),
-        otherwise: yup.date().nullable()
-    }
+    [ClinicalDetailsFields.IS_SYMPTOMS_DATE_UNKNOWN]: yup.boolean().required(),
+    [ClinicalDetailsFields.SYMPTOMS_START_DATE]: yup.date().when([ClinicalDetailsFields.IS_SYMPTOMS_DATE_UNKNOWN, ClinicalDetailsFields.DOES_HAVE_SYMPTOMS], {
```

Comment:
In this current form, if `doesHaveSymptoms` is false then the date is still required which it shouldn't be. You need to use the function format of yup's `when` API- then you'll be able to have three schema options.

Reply:
fixed

Reply:
I don't see that you fixed it you only removed the `=== true` which is not the problem I brought up.